pkg/apis/machine/v1beta1/machineset_webhook.go

```diff
@@ -0,0 +1,147 @@
+package v1beta1
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	osconfigv1 "github.com/openshift/api/config/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// machineSetValidatorHandler validates MachineSet API resources.
+// implements type Handler interface.
+// https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/webhook/admission#Handler
+type machineSetValidatorHandler struct {
+	clusterID         string
```

Comment:
since this is now used by 4 types it could be a struct on its own that we embed?

Reply:
Do you mean the clusterID, webhookOperations and decoder options?

Reply:
yep, wdyt?

Reply:
Sounds good, I'll take a look